games-strategy/freeciv/freeciv-2.6.0.recipe

```diff
@@ -0,0 +1,123 @@
+DESCRIPTION="Freeciv is a free Civilization-like game, primarily \
+for X under Unix. It has support for multiplayer games locally or \
+over a network, and an AI which gives most people a run for their money.
+Freeciv aims to be mostly rule-compatible with Civilization II [tm], \
+published by Sid Meier and Microprose [tm]. A few rules are different \
+where we think it makes more sense, and we have lots and lots of \
+adjustable parameters to make customizing games possible.
+Freeciv has been implemented completely independently of Civilization; \
+you do not need to own Civilization to play Freeciv."
+HOMEPAGE="http://www.freeciv.org/"
+COPYRIGHT="1996-2018 The Freeciv Team"
+LICENSE="GNU GPL v2"
+REVISION="1"
+SOURCE_URI="https://netix.dl.sourceforge.net/project/freeciv/Freeciv%202.6/$portVersion/freeciv-$portVersion.tar.bz2"
+CHECKSUM_SHA256="7c20399198d6c7d846fed9a69b02e01134ae5340a3ae0f99d1e38063ade6c999"
+PATCHES="freeciv-$portVersion.patchset"
+ADDITIONAL_FILES="freeciv.rdef.in"
+
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+commandSuffix=$secondaryArchSuffix
+commandBinDir=$binDir
+if [ -z "${ARCHITECTURES##*!$targetArchitecture*}" ]; then
+	commandSuffix=
+	commandBinDir=$prefix/bin
+fi
+
+GLOBAL_WRITABLE_FILES="
+	settings/freeciv/database.lua keep-old
+	"
+
+PROVIDES="
+	freeciv$secondaryArchSuffix = $portVersion
+	app:FreeCiv = $portVersion
+	cmd:freeciv_manual$commandSuffix
+	cmd:freeciv_sdl$commandSuffix
+	cmd:freeciv_server$commandSuffix
+	lib:libfreeciv$secondaryArchSuffix
+	lib:libfreeciv_srv$secondaryArchSuffix
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libbz2$secondaryArchSuffix
+	lib:libcurl$secondaryArchSuffix
+	lib:libcrypto$secondaryArchSuffix
+	lib:libfreetype$secondaryArchSuffix
+	lib:libGL$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
+	lib:libintl$secondaryArchSuffix
+	lib:liblzma$secondaryArchSuffix
+	lib:libreadline$secondaryArchSuffix
+	lib:libsdl_1.2$secondaryArchSuffix
+	lib:libsdl_gfx$secondaryArchSuffix
+	lib:libsdl_image_1.2$secondaryArchSuffix
+	lib:libsdl_mixer_1.2$secondaryArchSuffix
+	lib:libsdl_ttf_2.0$secondaryArchSuffix
+	lib:libssl$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libbz2$secondaryArchSuffix
+	devel:libcurl$secondaryArchSuffix
+	devel:libfreetype$secondaryArchSuffix
+	devel:libiconv$secondaryArchSuffix
+	devel:libintl$secondaryArchSuffix
+	devel:liblzma$secondaryArchSuffix
+	devel:libreadline$secondaryArchSuffix
+	devel:libsdl$secondaryArchSuffix
+	devel:libsdl_gfx$secondaryArchSuffix
+	devel:libsdl_image$secondaryArchSuffix
+	devel:libsdl_mixer$secondaryArchSuffix
+	devel:libsdl_ttf$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:aclocal
+	cmd:autoconf
+	cmd:find
+	cmd:gcc$secondaryArchSuffix
+	cmd:libtoolize$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	"
+
+BUILD()
+{
+	autoreconf -vfi
```

Comment:
Suggestion:
```diff
-	autoreconf -vfi
```
Seems it's not needed, with that you could enable sqlite3 and MagickWand (with a small patch in the configure file) :)

Reply:
@Begasus @extrowerk I have Lua working :)

Reply:
@Begasus I think autoreconf is useful for getting missing dependencies. Without it, I kept getting `aclocal-1.14 is missing from your computer`.

Also, on the recipe, do I add ImageMagick as `lib:libimagemagick` and `devel:libimagemagick`, and sqlite3 as `lib:libsqlite` and `devel:libsqlite`?

Reply:
@Begasus:
We could use
* `--enable-fcdb=no/all/mysql/postgres/sqlite3`
* `--with-sqlite3-prefix=PFX`: Prefix where SQLite3 is installed
* `--with-magickwand=DIR`: Imagemagick installation directory.

I think we can use these instead of adding libs and devel. What do you think?

Reply:
first one is needed to set sqlite3 :) pkg-config makes sure the packages (lib + devel) are found, there is just a check for the header in configure were the path isn't correct as used by imagemagic

Reply:
My bad, configure is not included in the source, so you do need autoreconf :)

Reply:
I think I'm lost :o 
I did set `--enable-fcdb` to `sqlite3`, but I have no idea what to do from there.
I looked in `configure` but no success :(

Reply:
--enable-fcmp=sqlite3 is right, just need to add it to the requirements then too (lib + devel), at the end of configure you'll get a summarize where it shows you it is enabled

Reply:
Looks like there is some problem with libsqlite3 and the beta, will proceed without it